File: backend/schemas/strategy_dev.py
from pydantic import BaseModel, Field


class StrategyFileCreate(BaseModel):
    file_name: str = Field(min_length=1)
    strategy_name: str = "新建策略"
    description: str = "请填写策略说明"


class StrategyImportRequest(BaseModel):
    file_name: str = Field(min_length=1)
    code_content: str = Field(min_length=1)


class StrategyStatusUpdate(BaseModel):
    status: str = Field(pattern="^(enabled|disabled)$")


class StrategyFileRecord(BaseModel):
    id: int
    file_name: str
    file_path: str
    strategy_name: str
    version: str
    description: str
    status: str
    last_modified_at: str | None = None
    last_run_at: str | None = None
    created_at: str
    today_signal_count: int = 0


class StrategyContent(BaseModel):
    strategy_id: int
    file_name: str
    code_content: str


class StrategyContentUpdate(BaseModel):
    code_content: str = Field(min_length=1)
    remark: str | None = None


class StrategyValidationResult(BaseModel):
    valid: bool
    message: str
    technical_detail: str | None = None
    strategy_name: str | None = None
    version: str | None = None
    description: str | None = None


class StrategyRunRecord(BaseModel):
    id: int
    run_id: str
    strategy_id: int
    strategy_name: str = ""
    strategy_file_name: str = ""
    strategy_version: str = ""
    strategy_code_hash: str = ""
    task_id: str
    status: str
    signal_count: int
    started_at: str | None = None
    finished_at: str | None = None
    message: str
    technical_detail: str | None = None


class StrategySignalRecord(BaseModel):
    id: int
    strategy_id: int
    run_id: str
    strategy_name: str
    symbol: str
    name: str
    action: str
    price: float
    amount: float | None = None
    reason: str
    status: str
    signal_time: str
    order_id: str | None = None
    created_at: str


class SignalStatusUpdate(BaseModel):
    status: str = Field(pattern="^(未处理|已下单|已忽略|已过期)$")


class StrategyVersionRecord(BaseModel):
    id: int
    strategy_id: int
    version_no: str
    code_hash: str
    remark: str | None = None
    created_at: str


class StrategyVersionDetail(StrategyVersionRecord):
    code_content: str


class StrategyVersionCompare(BaseModel):
    left_version_id: int
    right_version_id: int
    left_content: str
    right_content: str

File: frontend/src/types/strategyDev.ts
import type { PageResult } from './api';

export interface StrategyFileRecord {
  id: number;
  file_name: string;
  file_path: string;
  strategy_name: string;
  version: string;
  description: string;
  status: string;
  last_modified_at?: string | null;
  last_run_at?: string | null;
  created_at: string;
  today_signal_count: number;
}

export interface StrategyContent {
  strategy_id: number;
  file_name: string;
  code_content: string;
}

export interface StrategyValidationResult {
  valid: boolean;
  message: string;
  technical_detail?: string | null;
  strategy_name?: string | null;
  version?: string | null;
  description?: string | null;
}

export interface StrategyRunRecord {
  id: number;
  run_id: string;
  strategy_id: number;
  strategy_name?: string;
  strategy_file_name?: string;
  strategy_version?: string;
  strategy_code_hash?: string;
  task_id: string;
  status: string;
  signal_count: number;
  started_at?: string | null;
  finished_at?: string | null;
  message: string;
  technical_detail?: string | null;
}

export interface StrategySignalRecord {
  id: number;
  strategy_id: number;
  run_id: string;
  strategy_name: string;
  symbol: string;
  name: string;
  action: string;
  price: number;
  amount?: number | null;
  reason: string;
  status: string;
  signal_time: string;
  created_at: string;
}

export interface StrategyVersionRecord {
  id: number;
  strategy_id: number;
  version_no: string;
  code_hash: string;
  remark?: string | null;
  created_at: string;
}

export interface StrategyVersionDetail extends StrategyVersionRecord {
  code_content: string;
}

export type StrategyFilePage = PageResult<StrategyFileRecord>;
export type StrategyRunPage = PageResult<StrategyRunRecord>;
export type StrategySignalPage = PageResult<StrategySignalRecord>;
export type StrategyVersionPage = PageResult<StrategyVersionRecord>;
